packages/typescript-estree/src/convert.ts

```diff
@@ -697,7 +697,7 @@ export class Converter {
         return this.createNode<TSESTree.CatchClause>(node, {
           type: AST_NODE_TYPES.CatchClause,
           param: node.variableDeclaration
-            ? this.convertChild(node.variableDeclaration.name)
+            ? this.convertChild(node.variableDeclaration).id
```

Comment:
now that `catch (x: any)` is valid syntax, could you also add some tests for this?
Just add a few into this folder so we can validate that the type annotation is generated correctly.

https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/shared-fixtures/fixtures/typescript/basics

Reply:
Done. It now throws 

```
@typescript-eslint/typescript-estree: FAIL tests/ast-alignment/spec.ts
@typescript-eslint/typescript-estree:   ● TEST FAIL [BABEL ERRORED, BUT TS-ESTREE DID NOT] - /home/tehdog/data/dev/2020/typescript-eslint/packages/shared-fixtures/fixtures/typescript/basics/catch-type-any.src.ts
@typescript-eslint/typescript-estree:     expect(received).toEqual(expected) // deep equality
@typescript-eslint/typescript-estree:     Expected: [SyntaxError: Unexpected token, expected ")" (3:10)
@typescript-eslint/typescript-estree:       1 | try {
@typescript-eslint/typescript-estree:       2 |·
@typescript-eslint/typescript-estree:     > 3 | } catch (e: any) {
@typescript-eslint/typescript-estree:         |           ^
@typescript-eslint/typescript-estree:       4 |·
@typescript-eslint/typescript-estree:       5 | }
@typescript-eslint/typescript-estree:       6 | ]
@typescript-eslint/typescript-estree:     Received: null
@typescript-eslint/typescript-estree:       33 |     if (!typeScriptESTreeResult.parseError) {
@typescript-eslint/typescript-estree:       34 |       it(`TEST FAIL [BABEL ERRORED, BUT TS-ESTREE DID NOT] - ${filename}`, () => {
@typescript-eslint/typescript-estree:     > 35 |         expect(typeScriptESTreeResult.parseError).toEqual(
@typescript-eslint/typescript-estree:          |                                                   ^
@typescript-eslint/typescript-estree:       36 |           babelParserResult.parseError,
@typescript-eslint/typescript-estree:       37 |         );
@typescript-eslint/typescript-estree:       38 |       });
@typescript-eslint/typescript-estree:       at Object.<anonymous> (tests/ast-alignment/spec.ts:35:51)
@typescript-eslint/typescript-estree:   ● TEST FAIL [BABEL ERRORED, BUT TS-ESTREE DID NOT] - /home/tehdog/data/dev/2020/typescript-eslint/packages/shared-fixtures/fixtures/typescript/basics/catch-type-unknown.src.ts
@typescript-eslint/typescript-estree:     expect(received).toEqual(expected) // deep equality
@typescript-eslint/typescript-estree:     Expected: [SyntaxError: Unexpected token, expected ")" (3:10)
@typescript-eslint/typescript-estree:       1 | try {
@typescript-eslint/typescript-estree:       2 |·
@typescript-eslint/typescript-estree:     > 3 | } catch (e: unknown) {
@typescript-eslint/typescript-estree:         |           ^
@typescript-eslint/typescript-estree:       4 |·
@typescript-eslint/typescript-estree:       5 | }
@typescript-eslint/typescript-estree:       6 | ]
@typescript-eslint/typescript-estree:     Received: null
@typescript-eslint/typescript-estree:       33 |     if (!typeScriptESTreeResult.parseError) {
@typescript-eslint/typescript-estree:       34 |       it(`TEST FAIL [BABEL ERRORED, BUT TS-ESTREE DID NOT] - ${filename}`, () => {
@typescript-eslint/typescript-estree:     > 35 |         expect(typeScriptESTreeResult.parseError).toEqual(
@typescript-eslint/typescript-estree:          |                                                   ^
@typescript-eslint/typescript-estree:       36 |           babelParserResult.parseError,
@typescript-eslint/typescript-estree:       37 |         );
@typescript-eslint/typescript-estree:       38 |       });
@typescript-eslint/typescript-estree:       at Object.<anonymous> (tests/ast-alignment/spec.ts:35:51)
```

which i guess is good?

Reply:
yup that's to be expected; babel doesn't support it, and babel throws for things it doesn't support

You can ignore the fixture you added from the babel AST alignment tests here:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts

Reply:
i've added them to the ignored fixtures